moz_security/sandboxes/heka/output/moz_security_hh_ip_reputation.lua

```diff
@@ -0,0 +1,322 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+#  Mozilla Security Heavy Hitters Monitor to IP Reputation Output Plugin
+
+See: https://cacm.acm.org/magazines/2009/10/42481-finding-the-frequent-items-in-streams-of-data/abstract
+
+Computes heavy hitters by IP and send them to the IP Reputation Service (tigerblood) with HTTP with Hawk auth.
+
+## Sample Configuration
+```lua
+filename = "moz_security_hh_ip_reputation.lua"
+message_matcher = "Logger == 'input.nginx'"
+ticker_interval = 60
+preserve_data = true
+
+id_field = "Fields[remote_addr]"
+-- id_field_capture = ",? *([^,]+)$"a -- optional e.g. extract the last entry in a comma delimited list
+
+-- hh_items = 10000 -- optional, defaults to 10000 (maximum number of heavy hitter IDs to track), NB: should be <= tigerblood's MAX_ENTRIES setting
+
+-- config for publishing to IP Reputation
+config = {
+   base_url = "https://tigerblood.prod.mozaws.net", -- NB: no trailing slash
+   id = "fxa_heavy_hitters", -- hawk ID
+   key = "8eb54d0469ef000f8819a248c3f4465dc506bb2ad339aaa09466644b2214ffc8", -- hawk secret
```

Comment:
Name secrets with an underscore `_key` so the are not displayed in the view source UI.